src/command_modules/azure-cli-interactive/azclishell/az_completer.py

```diff
@@ -130,11 +130,12 @@ def __init__(self, commands, global_params=True, outstream=sys.stderr):
 
     def validate_completion(self, param, words, text_before_cursor, double=True):
         """ validates that a param should be completed """
-        return param.lower().startswith(words.lower()) and param.lower() != words.lower() and\
-            param not in text_before_cursor.split() and not \
-            text_before_cursor[-1].isspace() and\
-            (not (double and param in self.same_param_doubles) or
-             self.same_param_doubles[param] not in text_before_cursor.split())
+        result = param.lower().startswith(words.lower()) and param.lower() != words.lower()
+        result = result and param not in text_before_cursor.split() and not text_before_cursor[-1].isspace()
```

Comment:
Split the logic expression into meaningful parts. Assign each expression to its own variable. Make the variable name meaningful. Combine all the variable in a final logic expression.